AsyncDisplayKit/Details/ASCollectionDataController.mm

```diff
@@ -28,22 +28,30 @@ @implementation ASCollectionDataController {
   NSMutableDictionary *_pendingIndexPaths;
 }
 
-- (void)prepareForReloadData
+- (instancetype)initWithAsyncDataFetching:(BOOL)asyncDataFetchingEnabled
 {
-  _pendingNodes = [NSMutableDictionary dictionary];
-  _pendingIndexPaths = [NSMutableDictionary dictionary];
+  self = [super initWithAsyncDataFetching:asyncDataFetchingEnabled];
+  if (self != nil) {
+    _pendingNodes = [NSMutableDictionary dictionary];
+    _pendingIndexPaths = [NSMutableDictionary dictionary];
+  }
+  return self;
+}
 
-  [[self supplementaryKinds] enumerateObjectsUsingBlock:^(NSString *kind, NSUInteger idx, BOOL *stop) {
+- (void)prepareForReloadData
```

Comment:
Doesn't this potentially get called multiple times, and need to recreate the _pendingNodes and _pendingIndexPaths dictionaries?